kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs

```diff
@@ -653,8 +651,6 @@ impl<'tcx> GotocCtx<'tcx> {
                 let layout = self.layout_of(dst_mir_ty);
                 if layout.is_zst() || dst_type.sizeof_in_bits(&self.symbol_table) == 0 {
                     // We ignore assignment for all zero size types
-                    // Ignore generators too for now:
-                    // https://github.com/model-checking/kani/issues/416
```

Comment:
I believe this comment was outdated because the if condition only checks for ZST.